src/main/java/mate/jdbc/dao/impl/CarDaoImpl.java

```diff
@@ -0,0 +1,210 @@
+import mate.jdbc.util.ConnectionUtil;
+
+@Dao
+public class CarDaoImpl implements CarDao {
+    @Override
+    public Car create(Car car) {
+        String insertRequest = "INSERT INTO cars (model, manufacturer_id) VALUES (?, ?)";
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement statement = connection
+                        .prepareStatement(insertRequest, Statement.RETURN_GENERATED_KEYS)) {
+            statement.setString(1, car.getModel());
+            statement.setLong(2, car.getManufacturer().getId());
+            statement.executeUpdate();
+            ResultSet geneartedKeys = statement.getGeneratedKeys();
+            if (geneartedKeys.next()) {
+                Long id = geneartedKeys.getObject(1, Long.class);
+                car.setId(id);
+            }
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can't insert " + car, e);
+        }
+        insertDrivers(car);
+        return car;
+    }
+
+    @Override
+    public Optional<Car> get(Long id) {
+        String selectRequesr = "SELECT c.id as car_id, model, "
+                + "m.id as manufacturer_id, m.name, m.country "
+                + "FROM cars c "
+                + "JOIN manufacturers m ON c.manufacturer_id = m.id "
+                + "WHERE c.id = ? and c.is_deleted = FALSE";
+        Car car = null;
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement statement = connection.prepareStatement(selectRequesr)) {
+            statement.setLong(1, id);
+            ResultSet resultSet = statement.executeQuery();
+            if (resultSet.next()) {
+                car = getCar(resultSet);
+            }
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can't get car by id " + id, e);
+        }
+        Optional<Car> carOptional = Optional.ofNullable(car);
+        if (carOptional.isPresent()) {
+            car.setDrivers(getDriversForCar(id));
+        }
+        return carOptional;
+    }
+
+    @Override
+    public List<Car> getAll() {
+        String selectRequesr = "SELECT c.id as car_id, model, "
+                + "m.id as manufacturer_id, m.name, m.country "
+                + "FROM cars c "
+                + "JOIN manufacturers m ON c.manufacturer_id = m.id "
+                + "WHERE c.is_deleted = FALSE";
+        List<Car> cars = new ArrayList<>();
+        Car car = null;
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement statement = connection.prepareStatement(selectRequesr)) {
+            ResultSet resultSet = statement.executeQuery();
+            while (resultSet.next()) {
+                car = getCar(resultSet);
+                cars.add(car);
+            }
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can't get cars from DB", e);
+        }
+        for (Car car1: cars) {
```

Comment:
```suggestion
        for (Car currentCar: cars) {
```

It's better to name something like this, but do not use c or a name with a number